### ISSUE TITLE
Constant training loss observed when using mpt-7b_dolly_sft.yaml config

### ISSUE THREAD

**Issue by suehyunpark** [mosaicml/llm-foundry#290]
Hello, I'm trying to fine-tune MPT-7B starting from `mpt-7b_dolly_sft.yaml`,
but I observe that train loss, cross entropy, and perplexity are all fixed at a constant value throughout the entire training phase.   
I used the specified `mosaicml/dolly_hhrlhf` dataset and reduced `max_seq_len` to 512 for test runs using 2 x A100-80GB GPUs. Other than that and adding a wandb logger, I didn't make changes to the yaml file.

<details>
  <summary>Click to see my `mpt-7b_dolly_sft.yaml`</summary>
```

    max_seq_len: 2048
    global_seed: 17
    
    # Run Name
    run_name: # If left blank, will be read from env var $RUN_NAME
    
    model:
      name: hf_causal_lm
      pretrained: true
      pretrained_model_name_or_path: mosaicml/mpt-7b
      config_overrides:
        attn_config:
          attn_impl: triton
          # Set this to `true` if using `train_loader.dataset.packing_ratio` below
          attn_uses_sequence_id: false
    
    # Tokenizer
    tokenizer:
      name: mosaicml/mpt-7b
      kwargs:
        model_max_length: ${max_seq_len}
    
    
    # Dataloaders
    train_loader:
      name: finetuning
      dataset:
        hf_name: mosaicml/dolly_hhrlhf
        split: train
        max_seq_len: ${max_seq_len}
        allow_pad_trimming: false
        decoder_only_format: true
      # # Use `python llmfoundry/data/packing.py --yaml-path /path/to/this/yaml/ ...`
      # # to profile this run's optimal packing_ratio as it depends on GPU count,
      # # batch size, sequence length
      # packing_ratio:
      shuffle: true
    drop_last: true
    num_workers: 8
    pin_memory: false
    prefetch_factor: 2
    persistent_workers: true
    timeout: 0
  
    eval_loader:
      name: finetuning
      dataset:
        hf_name: mosaicml/dolly_hhrlhf
        split: test
        max_seq_len: ${max_seq_len}
        allow_pad_trimming: false
        decoder_only_format: true
        # packing_ratio:
        shuffle: true
      drop_last: true
      num_workers: 8
      pin_memory: false
      prefetch_factor: 2
      persistent_workers: true
      timeout: 0
  
    # Optimization
    scheduler:
      name: linear_decay_with_warmup  # linear no warmup is HF default which dolly used
      t_warmup: 50ba  # add some warmup though, seems to help with MPT
      alpha_f: 0
    
    optimizer:
      # Based on Dolly
      name: decoupled_adamw
      lr: 5.0e-6
      betas:
      - 0.9
      - 0.999
      eps: 1.0e-8
      weight_decay: 0
    
    algorithms:
      gradient_clipping:
        clipping_type: norm
        clipping_threshold: 1.0
    
    max_duration: 2ep  # 2-3 epochs seems like the sweet spot
    eval_interval: 1ep
    # eval_subset_num_batches: -1
    eval_first: true
    global_train_batch_size: 48  # somewhere in the 6-8 * numgpus range seems good
    
    # System
    seed: ${global_seed}
    device_eval_batch_size: 8
    device_train_microbatch_size: 8
    # device_train_microbatch_size: auto
    precision: amp_bf16
    
    # FSDP
    fsdp_config:
      sharding_strategy: FULL_SHARD
      mixed_precision: PURE
      activation_checkpointing: true
      activation_checkpointing_reentrant: false
      activation_cpu_offload: false
      limit_all_gathers: true
      verbose: false
    
    # Logging
    progress_bar: false
    log_to_console: true
    console_log_interval: 1ba
    
    callbacks:
      speed_monitor:
        window_size: 10
      lr_monitor: {}
      memory_monitor: {}
      runtime_estimator: {}
    
    loggers:
      wandb:
        project: prodchat
        name: ${run_name}
        entity: suehyun
    
    # Checkpoint to local filesystem or remote object store
    # save_interval: 5000ba
    # save_num_checkpoints_to_keep: 1  # Important, this cleans up checkpoints saved to DISK
    # save_folder: ./{run_name}/checkpoints
    # save_folder: s3://my-bucket/my-folder/{run_name}/checkpoints

```  ```
</details>





```
	 Train loss/train/total: 10.8125
	 Train metrics/train/LanguageCrossEntropy: 10.8125
	 Train metrics/train/LanguagePerplexity: 49637.4062
```
<img width="807" alt="image" src="https://github.com/mosaicml/llm-foundry/assets/77909639/6f960332-0b22-4cc0-be1b-c32789d5c21b">  

Tweaking some hyperparameters, e.g., learning rate or using a custom dataset didn't fix the problem.   
<img width="1392" alt="image" src="https://github.com/mosaicml/llm-foundry/assets/77909639/9e375222-84da-43ec-aabf-1ec55a5cc77f">  

 
I'm aware of previous discussions (#58 #70) that the provided yaml examples are not golden, but it's really strange that the configuration is not even working for a minimal example.  

Has anyone encountered a same issue? I'd really appreciate any comments or advices.  
Also, if this helps, I used the `mosaicml/llm-foundry:1.13.1_cu117-latest` image as the base image and followed the installation steps in the provided Dockerfile.

Thanks in advance!

**Comment by bratao** [comment]
+1 that is really frustrating that something such as the used config can´t be shared that you are not a customer. They should advertise llm-foundry as open-core and not open-source.

**Comment by growlix** [comment]
Hi @suehyunpark, currently trying to repro this!

**Comment by growlix** [comment]
So I attempted to reproduce your issue by pulling the `mosaicml/llm-foundry:1.13.1_cu117-latest` image and executing the following commands:

```
git clone git@github.com:mosaicml/llm-foundry.git
cd llm-foundry/scripts/train
composer train.py suehyunpark.yaml
```

where `suehyunpark.yaml` is the yaml you provided above. This error'd the first time because there is an indentation error in the yaml you provided: within the `train_loader` section of the yaml, the fields from `drop_last` to `timeout` are under-indented. After fixing this, I reran `composer train.py suehyunpark.yaml` and the model trained without issue (see attached screenshot):

<img width="1054" alt="Screenshot 2023-06-08 at 4 31 23 PM" src="https://github.com/mosaicml/llm-foundry/assets/11728034/947e9096-3411-4a28-9994-fa7d4e5a918b">

Please let me know if there are additional details that could help me reproduce your issue.

*UPDATE* I noticed you said you changed `max_seq_len` from 2048 to 512 in your yaml, so I reran with `max_seq_len=512` and everything still worked fine.

<img width="1066" alt="Screenshot 2023-06-08 at 4 46 22 PM" src="https://github.com/mosaicml/llm-foundry/assets/11728034/880268ce-6d9e-4841-a758-a950d6777ea2">

**Comment by suehyunpark** [comment]
Oh now I see where the problem is coming from 😅 Funny that I didn't notice the exact same fields from `drop_last` to `timeout` under `train_loader` were actually properly indented under `eval_loader`. I'll make sure to double check the formatting of config files next time.
Thanks a lot for your precious time reproducing my issue!

**Comment by vionwinnie** [comment]
This is very helpful! Thank you
